src/garage/sampler/worker_factory.py

```diff
@@ -0,0 +1,80 @@
+"""Worker factory used by Samplers to construct Workers."""
+import psutil
+
+from garage.sampler.worker import DefaultWorker
+
+
+def identity_function(value):
+    """Do nothing.
+
+    This function exists so it can be pickled.
+    """
+    return value
+
+
+class WorkerFactory:
+    """Constructs workers for Samplers.
+
+    The intent is that this object should be sufficient to avoid subclassing
+    the sampler. Instead of subclassing the sampler for e.g. a specific
+    backend, implement a specialized WorkerFactory (or specify appropriate
+    functions to this one). Not that this object must be picklable, since it
+    may be passed to workers. However, its fields individually need not be.
+
+    All arguments to this type must be passed by keyword.
+
+    Args:
+        seed(int): The seed to use to intialize random number generators.
+        n_workers(int): The number of workers to use.
+        max_path_length(int): The maximum length paths which will be sampled.
+        worker_class(type): Class of the workers. Instances should implement
+            the Worker interface.
+
+    """
+
+    def __init__(
+            self,
+            *,  # Require passing by keyword.
+            seed,
+            max_path_length,
+            n_workers=psutil.cpu_count(logical=False),
+            worker_class=DefaultWorker):
+        self.seed = seed
+        self.n_workers = n_workers
+        self.max_path_length = max_path_length
+        self.worker_class = worker_class
+
+    def get_worker_broadcast(self, objs, preprocess=identity_function):
```

Comment:
i was initially confused by the name of this API because it suggested to me that it's going to return a `Worker`. perhaps it can just be `broadcast` or `broadcast_to_n_workers`?